seq2science/rules/qc.smk

```diff
@@ -494,12 +494,25 @@ rule multiBamSummary:
         """
 
 
+def get_plotCor_opts(lst):
+    opts = config["deeptools_plotcorrelation"]
+    n = len(lst)
+    if "--plotHeight" not in opts:
+        # default: 9.5 cm
+        opts += f" --plotHeight {max(9.5, n*0.3)}"
+    if "--plotWidth" not in opts:
+        # default: 11 cm
+        opts += f" --plotWidth {max(11, 1.5+n*0.3)}"
+    return opts
+
+
 rule plotCorrelation:
     """
     Calculate the correlation between bams with deepTools.
     """
     input:
-        rules.multiBamSummary.output,
+        unpack(fingerprint_multiBamSummary_input),
+        cordata=rules.multiBamSummary.output,
```

Comment:
Is this necessary?

Reply:
`fingerprint_multiBamSummary_input` contains a list of bam files, which is used in `get_plotCor_opts()` to determine the number of samples. 

This seemed like the easiest source to me. If you have a better idea I can change it!

Reply:
ah like that. Sure makes sense! :+1: